src/pke/examples/tckks-interactive-mp-bootstrapping-Chebyshev.cpp

```diff
@@ -0,0 +1,309 @@
+        cerr << __func__ << " - " << __FILE__ << ":" << __LINE__ << " IntMPBoot - Ctxt Chebyshev Failed: " << endl;
+        cerr << __func__ << " - " << __FILE__ << ":" << __LINE__ << " - is diff <= eps?: " << tmp << endl;
+    }
+    else {
+        std::cout << "SUCESSFUL Bootstrapping!\n";
+    }
+}
+
+void TCKKSCollectiveBoot(enum ScalingTechnique scaleTech);
+
+int main(int argc, char* argv[]) {
+    std::cout << "Interactive (3P) Bootstrapping Ciphertext [Chebyshev] (TCKKS) started ...\n";
+
+    // Same test with different rescaling techniques in CKKS
+    TCKKSCollectiveBoot(ScalingTechnique::FIXEDMANUAL);
+    TCKKSCollectiveBoot(ScalingTechnique::FIXEDAUTO);
+    TCKKSCollectiveBoot(ScalingTechnique::FLEXIBLEAUTO);
+    TCKKSCollectiveBoot(ScalingTechnique::FLEXIBLEAUTOEXT);
+
+    std::cout << "Interactive (3P) Bootstrapping Ciphertext [Chebyshev] (TCKKS) terminated gracefully!\n";
+
+    return 0;
+}
+
+// Demonstrate interactive multi-party bootstrapping for 3 parties
+// We follow Protocol 5 in https://eprint.iacr.org/2020/304, "Multiparty
+// Homomorphic Encryption from Ring-Learning-With-Errors"
+
+void TCKKSCollectiveBoot(enum ScalingTechnique scaleTech) {
+    if (scaleTech != ScalingTechnique::FIXEDMANUAL && scaleTech != ScalingTechnique::FIXEDAUTO &&
+        scaleTech != ScalingTechnique::FLEXIBLEAUTO && scaleTech != ScalingTechnique::FLEXIBLEAUTOEXT) {
+        std::string errMsg = "ERROR: Scaling technique is not supported!";
+        OPENFHE_THROW(config_error, errMsg);
+    }
+
+    CCParams<CryptoContextCKKSRNS> parameters;
+    // A. Specify main parameters
+    /*  A1) Secret key distribution
+	* The secret key distribution for CKKS should either be SPARSE_TERNARY or UNIFORM_TERNARY.
+	* The SPARSE_TERNARY distribution was used in the original CKKS paper,
+	* but in this example, we use UNIFORM_TERNARY because this is included in the homomorphic
+	* encryption standard.
+	*/
+    SecretKeyDist secretKeyDist = UNIFORM_TERNARY;
+    parameters.SetSecretKeyDist(secretKeyDist);
+
+    /*  A2) Desired security level based on FHE standards.
+	* In this example, we use the "NotSet" option, so the example can run more quickly with
+	* a smaller ring dimension. Note that this should be used only in
+	* non-production environments, or by experts who understand the security
+	* implications of their choices. In production-like environments, we recommend using
+	* HEStd_128_classic, HEStd_192_classic, or HEStd_256_classic for 128-bit, 192-bit,
+	* or 256-bit security, respectively. If you choose one of these as your security level,
+	* you do not need to set the ring dimension.
+	*/
+    parameters.SetSecurityLevel(HEStd_128_classic);
+
+    /*  A3) Scaling parameters.
+	* By default, we set the modulus sizes and rescaling technique to the following values
+	* to obtain a good precision and performance tradeoff. We recommend keeping the parameters
+	* below unless you are an FHE expert.
+	*/
+    usint dcrtBits = 59;
```

Comment:
Reduce it to 50. 59 only makes sense in the non-interactive CKKS bootstrapping.

Reply:
Fixed as suggested.